src/GobanCore.ts

```diff
@@ -2708,6 +2708,11 @@ export abstract class GobanCore extends TypedEventEmitter<Events> {
         });
     }
     public pass(): void {
+        if (this.mode === "conditional") {
```

Comment:
This is the extent of the functional changes.  The rest was just tests I wrote to help me understand the code.